dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -246,7 +253,7 @@ def matmul(x1, x2, out=None, **kwargs):
     x1_desc = dpnp.get_dpnp_descriptor(x1)
     x2_desc = dpnp.get_dpnp_descriptor(x2)
     if x1_desc and x2_desc and not kwargs:
-        if x1_desc.size != x2_desc.size:
+        if x1_desc.ndim != 2 or x2_desc.ndim != 2:
```

Comment:
all 1D or 3D goes to fallback.

Reply:
In c++ backend implementation GEMM from mkl is used, this function works with 2D matrices. Maybe I am wrong and code on cython layer can extend dimension for 1D, but I don't think that we are ready to work with common 3D case. Please suggest better condition.
In any way, previous condition sends to fallback anything with different sizes, for example matrices with shapes (2, 3) and (3,4) are going to fallback...